File: software/examples/vbnet/ExampleCallback.vb
Imports System
Imports Tinkerforge

Module ExampleCallback
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Industrial Counter Bricklet

    ' Callback subroutine for all counter callback
    Sub AllCounterCB(ByVal sender As BrickletIndustrialCounter, ByVal counter As Long())
        Console.WriteLine("Counter (Channel 0): " + counter(0).ToString())
        Console.WriteLine("Counter (Channel 1): " + counter(1).ToString())
        Console.WriteLine("Counter (Channel 2): " + counter(2).ToString())
        Console.WriteLine("Counter (Channel 3): " + counter(3).ToString())
        Console.WriteLine("")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim ic As New BrickletIndustrialCounter(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register all counter callback to subroutine AllCounterCB
        AddHandler ic.AllCounterCallback, AddressOf AllCounterCB

        ' Set period for all counter callback to 1s (1000ms)
        ic.SetAllCounterCallbackConfiguration(1000, True)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/delphi/ExampleCallback.pas
program ExampleCallback;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletIndustrialCounter;

type
  TExample = class
  private
    ipcon: TIPConnection;
    ic: TBrickletIndustrialCounter;
  public
    procedure AllCounterCB(sender: TBrickletIndustrialCounter;
                           const counter: TArray0To3OfInt64);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change XYZ to the UID of your Industrial Counter Bricklet }

var
  e: TExample;

{ Callback procedure for all counter callback }
procedure TExample.AllCounterCB(sender: TBrickletIndustrialCounter;
                                const counter: TArray0To3OfInt64);
begin
  WriteLn(Format('Counter (Channel 0): %d', [counter[0]]));
  WriteLn(Format('Counter (Channel 1): %d', [counter[1]]));
  WriteLn(Format('Counter (Channel 2): %d', [counter[2]]));
  WriteLn(Format('Counter (Channel 3): %d', [counter[3]]));
  WriteLn('');
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  ic := TBrickletIndustrialCounter.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Register all counter callback to procedure AllCounterCB }
  ic.OnAllCounter := {$ifdef FPC}@{$endif}AllCounterCB;

  { Set period for all counter callback to 1s (1000ms) }
  ic.SetAllCounterCallbackConfiguration(1000, true);

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.
